src/consensus/pbft/pbfttypes.h

```diff
@@ -61,7 +90,7 @@ namespace pbft
           auto success = p->commit(
             version,
             [&]() {
-              ccf::Store::Tx tx(version);
+              tx.set_reserved_version(version);
```

Comment:
This shouldn't happen.

Reply:
Removed `set_reserved_version`. We can now either commit a pre-prepare by creating a new reserved transaction or pass in an existing transaction that already has the pre-prepare and just commit